docs/release_notes/ibexa_dxp_v4.6.md

```diff
@@ -0,0 +1,466 @@
+
+Attach context data to both the Cart and its individual Cart Entries.
+This feature enhances the flexibility and customization of your e-commerce application,
+enabling you to associate additional information with your cart and its contents.
+By leveraging context data, such as promo codes or custom texts,
+you can tailor the e-commerce experience for your customers and enhance the capabilities of your application.
+
+For more information, see [Adding context data](https://doc.ibexa.co/en/master/commerce/cart/cart_api/#adding-context-data).
+
+### New features and improvements in Personalization
+
+#### Triggers
+
+Triggers are push messages delivered to end users.
+With triggers, store managers can increase the engagement of their visitors and customers by delivering recommendations straight to their devices or mailboxes.
+While they experience improved fulfillment of their needs, more engaged customers mean bigger income for the store.
+The feature requires that your organization exposes an endpoint that passes data to an internal message delivery system and supports the following use cases:
+
+- Inducing a purchase by pushing a message with cart contents or equivalents, when the customer's cart status remains unchanged for a set time.
+
+- Inviting a customer to come back to the site by pushing a message with recommendations, when they haven't returned to the site for a set time.
+
+- Reviving the customer's interest by pushing a message with products that are similar to the ones the customer has already seen.
+
+- Inducing a purchase by pushing a message when a price of the product from the customer's wishlist decreases.
+
+For more information, see [Email triggers](https://doc.ibexa.co/projects/userguide/en/4.5/personalization/triggers.md).
+
+#### Multiple attributes in recommendation computation
+
+With this feature, you get an option to combine several attribute types when computing recommendations.
+As a result, users can be presented with recommendations from an intersection of submodel results.
+
+For more information, see [Submodel parameters](recommendation_api.md#submodel-parameters) and [Submodels](https://doc.ibexa.co/projects/userguide/en/latest/personalization/recommendation_models/#submodels).
+
+#### New scenario filter
+
+Depending on a setting that you make when defining a scenario, the recommendation response can now include either product variants or base products only.
+This way you can deliver more accurate recommendations and avoid showing multiple variants of the same product to the client.
+
+For more information, see [Commerce-specific filters](https://doc.ibexa.co/projects/userguide/en/latest/personalization/filters/#commerce-specific-filters).
+
+## Other changes
+
+### Expression Language
+
+New `project_dir()` expression language function that allows you to reference current project directory.
+
+### Site Factory events
+
+Site Factory events have been moved from the `lib` directory to `contracts`.
```

Comment:
I think it would be better to mention namespaces, not directories - they were moved to `Ibexa\Contracts\SiteFactory\Events`. We might mention that it's done in a BC-safe manner.

POV ping @mikadamczyk

Reply:
It could be:
Site Factory events have been moved from the `Ibexa\SiteFactory\ServiceEvent\Events` namespace to the `Ibexa\Contracts\SiteFactory\Events` namespace, keeping a BC-safe manner.